internal/utils/cli/error_test.go

```diff
@@ -0,0 +1,119 @@
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+)
+
+func TestErrNew(t *testing.T) {
+	t.Run("should create an error with minimal details", func(t *testing.T) {
+		err := NewErr(errors.New("something bad happened"))
+
+		assert.Equal(t, cliErr{cause: errors.New("something bad happened")}, err)
+
+		t.Run("and be recognized as a usage hider", func(t *testing.T) {
+			var usageHider ErrUsageHider
+			assert.True(t, errors.As(err, &usageHider), fmt.Sprintf("%T is not a usage hider", err))
+			assert.False(t, usageHider.HideUsage(), "should not hide usage")
+		})
+
+		t.Run("and be recognized as a suggester", func(t *testing.T) {
+			var suggester ErrSuggester
+			assert.True(t, errors.As(err, &suggester), fmt.Sprintf("%T is not a suggester", err))
+			assert.True(t, len(suggester.Suggestions()) == 0, "should have no suggestions")
+		})
+
+		t.Run("and be recognized as a link referrer", func(t *testing.T) {
+			var linkReferrer ErrLinkReferrer
+			assert.True(t, errors.As(err, &linkReferrer), fmt.Sprintf("%T is not a link referrer", err))
+			assert.True(t, len(linkReferrer.ReferenceLinks()) == 0, "should have no links")
+		})
+	})
+
+	t.Run("should create an error with complete details", func(t *testing.T) {
+		err := NewErr(
+			errors.New("something bad happened"),
+			ErrNoUsage{}, ErrSuggestion{"suggestion"}, ErrReferenceLink{"ref.link"},
+		)
+
+		assert.Equal(t,
+			cliErr{
+				cause:          errors.New("something bad happened"),
+				hideUsage:      true,
+				suggestions:    []interface{}{"suggestion"},
+				referenceLinks: []interface{}{"ref.link"},
+			},
+			err,
+		)
+
+		t.Run("and be recognized as a usage hider", func(t *testing.T) {
+			var usageHider ErrUsageHider
+			assert.True(t, errors.As(err, &usageHider), fmt.Sprintf("%T is not a usage hider", err))
+			assert.True(t, usageHider.HideUsage(), "should hide usage")
+		})
+
+		t.Run("and be recognized as a suggester", func(t *testing.T) {
+			var suggester ErrSuggester
+			assert.True(t, errors.As(err, &suggester), fmt.Sprintf("%T is not a suggester", err))
```

Comment:
similarly should this assertion message be `"%T is a suggester"` instead of "not"?